pytorch_lightning/trainer/properties.py

```diff
@@ -485,6 +487,30 @@ def sanity_checking(self, val: bool) -> None:
         elif self.sanity_checking:
             self.state.stage = None
 
+    @property
+    def global_step(self) -> int:
+        return self.train_loop.global_step
+
+    @property
+    def current_epoch(self) -> int:
+        return self.train_loop.current_epoch
+
+    @property
+    def max_epochs(self) -> Optional[int]:
+        return self.train_loop.max_epochs
+
+    @property
+    def min_epochs(self) -> Optional[int]:
+        return self.train_loop.min_epochs
+
+    @property
+    def max_steps(self) -> Optional[int]:
+        return self.train_loop.max_steps
+
+    @property
+    def min_steps(self) -> Optional[int]:
+        return self.train_loop.min_steps
```

Comment:
will this be deprecated in future?

Reply:
I don't think so. The user is allowed to access it and it's useful, but it should be read-only :)

Reply:
is `batch_idx` intentionally left out of these read-only properties?

Reply:
yes, it's not meant to be on the trainer because 1) it is accessible to the user through the hooks 2) it anyway wouldn't be clear what the meaning of this variable is since it is specific to the loop.